### DISCUSSION TITLE
[support question] How to find all my comments

### DISCUSSION THREAD

**Discussion by omega3** [community/community#58062] [New to GitHub] [answered]
Is there an option to list or search for all my comments written across various GitHub projects, usually on issues tabs?

**Comment by Gustavonn07** [comment]
Hey, how are you? I didn't understand your question very well, but if you're talking about commits, you can find them in this part here:

![image](https://github.com/community/community/assets/84361085/a9573c4d-e18d-4d4b-ad2a-acda3b5a0c29)

all updates made to your project and the comments you made when updating with commits (like `git commit -m "comment"`)
I'm sorry for my bad English, I hope I helped!

**Reply by IArnaut2** [reply]
THEY WANT TO VIEW THEIR COMMENTS ON ISSUES. Is that clear now?

**Reply by beedell-roke** [reply]
@IArnaut2, don't be rude.

**Comment by omega3** [comment]
I my issues I see Created, Assigned, Mentioned. What about all that I written but I didn't started, I am not assigned and were not mentioned? How to find all my comments in issues of any project?

**Reply by kakam360** [reply]
Great thanks for this❤

**Reply by githorse** [reply]
This should definitely be included in the "Filters" menu. I always think that "everything mentioning you" will do this, but it doesn't work.

**Comment by Gustavonn07** [comment, marked as answer]
To find all your comments in the issues of any project on GitHub, you can use the GitHub search functionality with a specific query. Here's how you can do it:

1. In the search bar at the top, enter the following query:

`is:issue commenter:your_username`
2. Replace your_username with your actual GitHub username. This query searches for all issues where you have commented.

3. Press Enter or click on the search icon.

GitHub will display a list of issues where you have commented, regardless of whether you created them, were assigned to them, or were mentioned in them.

You can further refine your search by adding additional filters to the query if you want to narrow down the results. For example, you can specify a particular repository or search for comments within a specific timeframe. Here's an example query that includes a repository filter:

`is:issue commenter:your_username repo:username/repository_name`
2. Replace your_username with your GitHub username and username/repository_name with the username and name of the repository where you want to search for comments.

Using these search queries, you should be able to find all your comments in the issues of any project on GitHub, even if you didn't create them, weren't assigned to them, and weren't mentioned in them.

example:

![image](https://github.com/community/community/assets/84361085/bcd21349-2b21-4a1b-b658-c246c597ffe0)

Hope this helps!

**Reply by rubyhcm** [reply]
It 's helpful!

**Reply by craigphicks** [reply]
`is:issue commenter:@me`

**Reply by rolodato** [reply]
Direct link: https://github.com/search?q=is%3Aissue%20commenter%3A%40me

**Reply by uuonda** [reply]
@rolodato It does not list issues where I'm authored. How do I combine this with `author:@me` filter? I want a single search query that would list every issue where I'm commenter or author.

**Reply by gersona** [reply]
@rolodato `author:@me OR commenter:@me` doesn't seem to work,
the closest I can think of is `involves:@me` , but it will also includes results mentions and assignees

**Reply by ghost** [reply]
Yeah but there should be a convenient UI option, this is atrociously bad UI / UX design as evidence by the thousands of views and thumbs ups on this thread.

**Reply by gerazenobi** [reply]
>  search for comments within a specific timeframe

Hi there, how do I do that ? (**note:** search issues where the **comment** was made within a timeframe)

Thanks in advance.

**Reply by elainajones** [reply]
Thanks! This helped me too. I wish they'd add a UI option. Commenting for visibility.

**Reply by realyukii** [reply]
is there a similar way for gists? I tried [this query](https://gist.github.com/search?q=commenter%3Arealyukii)  and it didn't work.

also tried `user:realyukii` but unfortunately, it only shows my gists not my comment on other ppl gists.

reference: [docs](https://docs.github.com/en/search-github/searching-on-github/searching-gists)

**Reply by YuZhang-GitHub** [reply]
Working perfectly. Thanks a lot!

**Reply by 0xfeeddeadbeef** [reply]
I tried to find a PR comment I made a week ago and it does not show up in search results.

**Reply by softyoda** [reply]
Could anyone add a button to do that? (idk affordance, i don't have to remember this command please...)

**Reply by IArnaut2** [reply]
How do I view my comments on other user's issues?

**Reply by rudi1602** [reply]
https://stackoverflow.com/a/43381243/14547215

**Reply by softyoda** [reply]
Please invade github HQ to ask msft to just have a sinmple button history instead of this nonsens

**Reply by jbartlett-p3** [reply]
Doesn't work for commit comments.

**Comment by Waldosek** [comment]
Hello, why can't I verify myself, what do I need to do to make it right, please answer urgently - Linea project, best regards

![Zrzut ekranu (88)](https://github.com/community/community/assets/136612350/e28c60af-431e-4ee1-b72f-16c2def02902)

**Reply by IPlayZed** [reply]
How is this related to this issue...?

**Reply by craigphicks** [reply]
@ IPlayZed
> How is this related to this issue...?

Spam is related to everything.  And nothing.

**Reply by IPlayZed** [reply]
It is OK, if OP of the comment needs help, but it should be in a new discussion.

**Reply by craigphicks** [reply]
It is advertising for some application "Linea".  Given that the first entry is "wallet connected" I would say it is probably malware.

**Comment by shodanx2** [comment]
This doesn't work ?

Here for example

![image](https://github.com/user-attachments/assets/6f6b20f1-daed-4975-b9fa-4c26e242300b)


but I just made a comment in an issue today, and it does not show up.

Example

![image](https://github.com/user-attachments/assets/1e70e105-8432-49c5-bd14-37bc4dc04b40)


This was posted 20 minutes ago and it does not show up in the search


I would like a place that just has all my lastest comments in it.

It should be right here

![image](https://github.com/user-attachments/assets/c7998d30-1f7e-4047-b019-b05e4f160b41)

**Comment by dlong500** [comment]
I have the same issue as @shodanx2.  Using `commenter:` or `involves:` without specifying a repo doesn't seem to show everything (only a partial list).  Not sure if there is some indexing that only gets updated once a week or something.  It would be nice to get clarification from the github folks on this matter.

For example, I have a comment I made 5 days ago in a particular repo that doesn't show up.  However, if I specify the `repo:` in the search it does show up.  My best guess would be that the indexes used for sitewide searches only get periodically refreshed.  I'm curious to test if the particular comment shows up when searching in a couple days.

**Reply by dlong500** [reply]
It does appear that there is a week or two delay before posts get indexed github-wide.  The comment I mentioned earlier is now showing in a global search using `commenter:@me`.  So that is something to bear in mind when trying to find content.  Also worth noting is that the  default "Best Match" sort seems to mostly just sort by newest to oldest based on the opening date of the issue (NOT your comment dates).  Changing the sort to "Recently updated" will likely give more useful results in the context of this discussion.

**Reply by profucius** [reply]
I haven't noticed it being that long of a delay. I just tried it myself, and I found a comment I made just an hour ago show up in my filter. You may want to change the filter order to "Recently updated" to see comments you've made in tickets that were recently updated. It's not technically when you updated it but when the ticket was, but it's pretty close.

**Comment by Visikde** [comment]
Go to your profile, scroll down, keep clicking "show more activity" until you find the post(s) you are looking for

**Comment by oldium** [comment]
This does not work for commit comments. How to find my comments to various code commits? They are even not shown in recent actvity.

![image](https://github.com/user-attachments/assets/d9390a70-5645-4ce5-910e-f496dfac2b3a)

**Comment by Marco1981s** [comment]
> To find all your comments in the issues of any project on GitHub, you can use the GitHub search functionality with a specific query. Here's how you can do it:
> 
> 
> 
> 1. In the search bar at the top, enter the following query:
> 
> 
> 
> `is:issue commenter:your_username`
> 
> 2. Replace your_username with your actual GitHub username. This query searches for all issues where you have commented.
> 
> 
> 
> 3. Press Enter or click on the search icon.
> 
> 
> 
> GitHub will display a list of issues where you have commented, regardless of whether you created them, were assigned to them, or were mentioned in them.
> 
> 
> 
> You can further refine your search by adding additional filters to the query if you want to narrow down the results. For example, you can specify a particular repository or search for comments within a specific timeframe. Here's an example query that includes a repository filter:
> 
> 
> 
> `is:issue commenter:your_username repo:username/repository_name`
> 
> 2. Replace your_username with your GitHub username and username/repository_name with the username and name of the repository where you want to search for comments.
> 
> 
> 
> Using these search queries, you should be able to find all your comments in the issues of any project on GitHub, even if you didn't create them, weren't assigned to them, and weren't mentioned in them.
> 
> 
> 
> example:
> 
> 
> 
> ![image](https://github.com/community/community/assets/84361085/bcd21349-2b21-4a1b-b658-c246c597ffe0)
> 
> 
> 
> Hope this helps!
> 
>

**Reply by Marco1981s** [reply]
Y

**Comment by burak-yildizoz** [comment]
I have the [same question](https://github.com/orgs/community/discussions/58062#discussioncomment-11426271) with @realyukii. Is it possible to search comments on gists?

**Comment by webdev23** [comment]
Cluttered and non functioning properly.

**Reply by judos** [reply]
For me it also doesn't work when search via query, or the autocomplete, with either "user:" or "commenter:" and also with either my username or "@me". Is it broken currently?

**Comment by gugs881** [comment]
> Is there an option to list or search for all my comments written across various GitHub projects, usually on issues tabs?

No, GitHub does not provide a direct option to list all your comments across issues. As a workaround, you can use the search with commenter:your-username or use the GitHub API.

**Comment by aherrick** [comment]
I built this to show my comments on issues/discussions and when that thread was last touched/updated. It's insane I had to build something custom for this. https://github.com/aherrick/GitHubActivityViewer

commenter:your-username only shows when the thread was last closed, etc. but thats worthless if the conversation continues on while "closed"

**Reply by shodanx2** [reply]
Thanks !
This works !

For example my latest comment 
https://github.com/Qalculate/libqalculate/issues/78#issuecomment-3422533566


Here is the result for commenter:shodanx2 plus SORT BY NEWEST
https://github.com/search?q=commenter%3Ashodanx2&type=discussions&s=date_created&o=desc

<img width="1824" height="1096" alt="image" src="https://github.com/user-attachments/assets/13d1def5-a8f5-46d9-a5b1-01bf3d6245bc" />

Here is the result of demo of [GitHubActivityViewer](https://github.com/aherrick/GitHubActivityViewer) by [aherrick](https://github.com/aherrick)

<img width="1696" height="1338" alt="image" src="https://github.com/user-attachments/assets/ed365642-03b5-4329-b31a-ea2ec2c1ca88" />

So as long as https://githubajh.z19.web.core.windows.net/ works, we have a solution at last !

**Comment by shuttittuppitt** [comment]
All of the comments (in this topic) from developers (or moderators or whatever title github gives such people) seem to be just workarounds. We appreciate those (when they work), but what we _really_ need is for our USER PROFILE PAGE (or whatever it's called) to have a section that's called "COMMENTS HISTORY".

I don't expect this change to be made any time soon, but it's really the ONLY way to let people do this, because these workarounds themselves require people to (somehow) 'know' about them (even though this lonely topic is the only part of Github that I'VE ever even _heard_ about them), & sometimes the workarounds _don't_ work.

**Reply by craigphicks** [reply]
A live video of a hunger striking coder might bring some visibility to the issue.

**Comment by bencutting164-cloud** [comment]
That’s a handy trick, I didn’t realize you could filter by commenter across issues. Makes it way easier to track past discussions without digging through repos manually. Thanks for sharing.

**Comment by AMDphreak** [comment]
This support question still maps to a product gap.

`commenter:` works in search, yet the Issues/PRs filter UI includes **Mentions** and not **Commenter** (or **Involves**). Discoverability suffers?people assume the capability does not exist when it is only missing from the dropdown.

I put together a concrete request covering that mismatch and hiring/shareable-link workflows: https://github.com/orgs/community/discussions/202870

Adding Commenter to the filter dropdowns would answer a lot of ?how do I find my comments?? threads like this.
